AirCasting/SearchAndFollow/SearchMap/SearchMapView.swift

```diff
@@ -0,0 +1,93 @@
+// Created by Lunar on 16/02/2022.
+//
+
+import CoreLocation
+import Foundation
+import SwiftUI
+import AirCastingStyling
+
+struct SearchMapView: View {
+    @StateObject var viewModel: SearchMapViewModel
+    @Environment(\.presentationMode) var presentationMode
+    
+    init(locationName: String, locationAddress: CLLocationCoordinate2D, parametr: String) {
+        _viewModel = .init(wrappedValue: .init(passedLocation: locationName, passedLocationAddress: locationAddress, passedParameter: parametr))
+    }
+    
+    var body: some View {
+        ZStack(alignment: .top, content: {
+            LoadingView(isShowing: $viewModel.showLoadingIndicator, activityIndicatorText: Strings.SearchMapView.loadingText) {
+                map
+            }
+            VStack(alignment: .center, content: {
+                textField
+                parametersText
+                redoneButton
+            })
+                .padding(.top, 50)
+                .padding(.horizontal)
+        })
+            .onChange(of: viewModel.shouldDismissView, perform: { _ in
+                viewModel.shouldDismissView ? self.presentationMode.wrappedValue.dismiss() : nil
```

Comment:
no need to check `viewModel.shouldDismissView` as it will be passed in as a parameter of this closure